hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioEventloop.java

```diff
@@ -36,57 +32,24 @@
 class NioEventloop extends Eventloop {
 
     final Selector selector = SelectorOptimizer.newSelector();
-    private final NioReactor reactor;
 
     NioEventloop(NioReactor reactor, NioReactorBuilder builder) {
         super(reactor, builder);
-        this.reactor = reactor;
     }
 
     @Override
     protected void run() throws Exception {
-        final NanoClock nanoClock = this.nanoClock;
```

Comment:
Revert. I put these fields in local variables for a reason.

Reply:
I guess this is true. But can you explain why you did it?

Reply:
I did some digging about this. I read [this](https://www.lenshood.dev/2021/06/21/why-assign-final-to-local-variable/), [this](https://stackoverflow.com/questions/2785964/in-arrayblockingqueue-why-copy-final-member-field-into-local-final-variable) and, [this](https://mail.openjdk.org/pipermail/core-libs-dev/2010-May/004165.html).

Benefits didn't seemed noticeable as they call this an "extreme optimization". So I prepared a really minor JMH:
```
@State(Scope.Benchmark)
public class MyBenchmark {
    private final int field = 1;

    @Benchmark
    public void forField(Blackhole blackhole) {
        for (int i = 0; i < 1_000_000_000; i++) {
            blackhole.consume(field);
        }
    }

    @Benchmark
    public void forLocal(Blackhole blackhole) {
        final int local = 1;
        for (int i = 0; i < 1_000_000_000; i++) {
            blackhole.consume(local);
        }
    }
}
```

And here is the result:
```
Benchmark              Mode  Cnt  Score   Error  Units
MyBenchmark.forField  thrpt   25  0.486 ± 0.002  ops/s
MyBenchmark.forLocal  thrpt   25  0.485 ± 0.001  ops/s
```

As you can see results are identical. Of course I am really new learning this stuff so my benchmark maybe is flawed. But anyway I would always love to learn more about these kind of stuff.

Reply:
It depends on the situation. I was also surprised about the impact. I was benchmarking the empty loop and saw that moving these fields to local variables made a big impact.